### DISCUSSION TITLE
zig 构建

### DISCUSSION THREAD

**Discussion by pyjcode** [zigcc/forum#106] [新手提问]
构建自己的程序让他在不依赖c系库和工具链的情况下运行，可能讲的不是很清楚，见谅，
因为自己也才开始做比较底层的开发

**Comment by jinzhongjia** [comment]
不要连接C库的情况下，就可以脱离C库，然后再指定对应的平台和架构，例如平台是freestanding,架构是x86,那么编译出来的东西就是可以在x86的裸机器上运行
但是问题在于zig没有那么多的纯zig库供你使用（现在不少库都是对c的包裹，例如我负责的[zig-webui](https://github.com/webui-dev/zig-webui)就是对c的包裹）

**Reply by pyjcode** [reply]
谢谢，您的意思是不是说，目前如果纯zig，很难实现一个比较有意义的工程，所以很多还是对c进行包裹

**Reply by jinzhongjia** [reply]
是的，所以不要排斥c，你也可以看看我在这里下面的评论，https://zigcc.github.io/zig-course/
zig是不可能替代c的，它更像是一个底层的工具，拓展c,包容c

**Reply by pyjcode** [reply]
唉，好吧，自己一直想要研究底层，但是c导致研究很困难，才开始学习zig的，这样来说的话，只能暂时接受这种情况，忽略库，也可以看明白建立在c库之上的东西

**Reply by jinzhongjia** [reply]
只要是底层，c 和 汇编 这两个东西无论如何都绕不过去的

**Reply by pyjcode** [reply]
恩，自己一直从事现代语言的开发，虽然大学学过c，不过，对于我来说，要去研读c的代码，确实有点困难，zig的出现就比较方便我这样的情况吧

**Comment by jinzhongjia** [comment]
我的观点是不要排斥c,c的缺点也就是缺少一定的安全检查，宏可能难以阅读，交叉编译苦难，包管理和构建系统混乱，抛开这些，c 可是一切操作系统的基石呢，还包括各式的嵌入式

**Reply by pyjcode** [reply]
好吧，谢谢解惑
